pages/gallery.tsx

```diff
@@ -1,3 +1,142 @@
+    const [imageUrlArray, setImageUrlArray] = useState([])
+    const [imageSrc, setImgSrc] = useState('')
+    const [open, setOpen] = useState(false)
+    const [nextImageSetUrl, setNextImageSetUrl] = useState('')
+    const getThis = async () => {
+        const url = ` https://graph.instagram.com/me/media?fields=id,media_url,timestamp,permalink&access_token=${process.env.NEXT_PUBLIC_IG_SECRET}`
+        try {
+            const data = await fetch(url)
+            const feed = await data.json()
+            console.log('feed :>> ', feed)
+            const imageUrls = feed.data.map(
+                (item: { media_url: any }) => item.media_url
+            )
+            if (feed.paging.next) {
+                setNextImageSetUrl(feed.paging.next)
+            }
+            setImageUrlArray(imageUrls)
+        } catch (error: any) {
+            console.error(error.message)
+        }
+    }
+
+    // Function for 25+ images
+    // const getThis = async () => {
+    //     const url = `https://graph.instagram.com/me/media?fields=id,media_url,timestamp,permalink&access_token=${process.env.NEXT_PUBLIC_IG_SECRET}`
+    //     let allMediaUrls = []
+
+    //     const fetchData = async nextUrl => {
+    //         try {
+    //             const response = await fetch(nextUrl || url)
+    //             const feed = await response.json()
+    //             const mediaUrls = feed.data.map(item => item.media_url)
+    //             allMediaUrls = allMediaUrls.concat(mediaUrls)
+
+    //             if (feed.paging && feed.paging.next) {
+    //                 // Fetch next page if available
+    //                 await fetchData(feed.paging.next)
+    //             } else {
+    //                 // All pages fetched, log the final result
+    //                 setImageUrls(allMediaUrls)
+    //             }
+    //         } catch (error) {
+    //             console.error(error.message)
+    //         }
+    //     }
+
+    //     await fetchData()
+    // }\
+
+    const onHelpMeGod = async () => {
+        try {
+            const data = await fetch(nextImageSetUrl)
+            const feed = await data.json()
+            const imageUrls = feed.data.map(
+                (item: { media_url: any }) => item.media_url
+            )
+
+            if (feed.paging.next) {
+                setNextImageSetUrl(feed.paging.next)
+            }
+
+            const validImageUrls = await Promise.all(
+                imageUrls.map(imageUrl => {
+                    return new Promise((resolve, reject) => {
+                        const img = new Image()
+                        img.src = imageUrl
+                        //I was deep . idk wtf this even does but it was the cmd v of a lifetime
+                        img.onload = () => resolve(imageUrl)
+                        img.onerror = () => resolve(null)
+                    })
+                })
+            )
+
+            const filteredImageUrls = validImageUrls.filter(
```

Comment:
Instead of doing this in two steps, just add the filter to the end of the map...if that works with the promises. i have no idea if it will, but lets see if god helps you here